demo/README.md

```diff
@@ -0,0 +1,203 @@
+# in-toto demo
+
+In this demo, we will use in-toto to secure a supply chain with a very simple
+workflow.
+Alice will be the project owner - she creates and signs the supply chain layout
+with her private key - and Bob and Carl will be the project functionaries - they
+carry out the steps of the supply chain as defined in the layout.
+
+For the sake of demonstrating in-toto, we will have you run all parts of the
+supply chain.
+This is, you will perform the commands on behalf of Alice, Bob and Carl as well
+as the client who verifies the final product.
+
+
+### Download and setup in-toto on *NIX (Linux, OS X, ..)
+```shell
+# Make sure you have git, python and pip installed on your system
+# and get in-toto
+git clone -b develop --recursive https://github.com/in-toto/in-toto.git
+
+# Change into project root directory
+cd in-toto
+
+# Install required packages (we strongly recommend using Virtual Environments)
+# http://docs.python-guide.org/en/latest/dev/virtualenvs/
+pip install -r requirements.txt
+
+# Install additional requirements that for some good reason are not in the
+# requirements file
+pip install pycrypto cryptography
+
+# Make sure in-toto is on your PYTHONPATH and used commands are on your PATH
+export PYTHONPATH=$PYTHONPATH:$PWD
+export PATH=$PATH:$PWD/toto
+
+# Change into the demo directoy and you are ready to start
+cd demo
+```
+Inside the demo directory you will find four directories: `owner_alice`,
+`functionary_bob`, `functionary_carl` and `final_product`. Alice, Bob and Carl
+already have RSA keys in each of their directories. This is what you see:
+```shell
+tree
+# the tree command gives you the following output
+# ├── README.md
+# ├── final_product
+# ├── functionary_bob
+# │   ├── bob
+# │   └── bob.pub
+# ├── functionary_carl
+# │   ├── carl
+# │   └── carl.pub
+# ├── owner_alice
+# │   ├── alice
+# │   └── alice.pub
+# └── run-demo.py
+```
+
+### Define supply chain layout (Alice)
+First, we will need to define the supply chain layout. To simplify this process,
+we provide a script that generates a simple layout for the purpose of the demo.
+In this supply chain layout, we have Alice, who is the project owner that creates
+the layout, Bob, who uses `vi` to create a Python program foo.py, and Carl, who
+uses `tar` to package up `foo.py` into a tarball which together with the in-toto
+metadata composes the final product that will eventually be installed and verified
+by the end user.
+
+```shell
+# Create and sign the supply chain layout on behalf of Alice
+cd owner_alice
+python create_layout.py
```

Comment:
I recommend that you set the general cryptography library (settings.GENERAL_CRYPTO_LIBRARY) to `pyca-cryptography`, which is faster and more solid than `pycrypto`.  We recently (past 2 weeks) set it to `pyca-cryptography` in TUF's settings, so that might be why it's not set here.